server/extensions/handler.go

```diff
@@ -0,0 +1,182 @@
+package extensions
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/gorilla/mux"
+	gocache "github.com/patrickmn/go-cache"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/time/rate"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	"github.com/argoproj/argo-cd/v2/pkg/client/clientset/versioned/typed/application/v1alpha1"
+	"github.com/argoproj/argo-cd/v2/server/rbacpolicy"
+	"github.com/argoproj/argo-cd/v2/util/rbac"
+)
+
+type item struct {
+	Resource string `json:"resource"`
+	Action   string `json:"action"`
+	Object   string `json:"object"`
+}
+
+var httpClient = &http.Client{
+	Timeout: 30 * time.Second,
```

Comment:
This seems like it should be configurable, potentially on a per-extension (and maybe even a per-endpoint) basis.